polaris-migrator/src/utilities/sass.ts

```diff
@@ -251,3 +259,50 @@ export function createInlineComment(text: string, options?: {prose?: boolean}) {
 
   return comment;
 }
+
+interface PluginOptions extends Options, NamespaceOptions {}
+
+interface PostCSSWalkers {
+  AtRule?: (node: AtRule) => void;
+  Comment?: (node: PostCSSComment) => void;
+  Declaration?: (node: Declaration) => void;
+  Rule?: (node: PostCSSRule) => void;
+}
+
+export function createSassMigrator(
+  name: string,
+  ruleFn: (options: PluginOptions) => PostCSSWalkers,
+) {
+  return (fileInfo: FileInfo, _: API, options: Options) => {
+    const plugin: Plugin = {
+      postcssPlugin: name,
+      // PostCSS will rewalk the AST every time a declaration/rule/etc is
+      // mutated by a plugin. This can be useful in some cases, but in ours we
+      // only want single-pass behaviour.
+      //
+      // This can be avoided in 2 ways:
+      //
+      // 1) Flagging each declaration as we pass it, then skipping it on
+      //    subsequent passes.
+      // 2) Using postcss's Once() plugin callback.
+      //
+      // We're going with the Once() callback as it's idomatic PostCSS.
+      Once(root: Root) {
+        // NOTE: We initialise the rule here _inside_ the postcss Once function
+        // so multiple passes can be performed without rules accidentally
+        // retaining scoped variables, etc.
+        const rules = ruleFn(options);
+
+        if (rules.AtRule) root.walkAtRules((node) => rules.AtRule!(node));
+        if (rules.Comment) root.walkComments((node) => rules.Comment!(node));
+        if (rules.Declaration)
+          root.walkDecls((node) => rules.Declaration!(node));
+        if (rules.Rule) root.walkRules((node) => rules.Rule!(node));
```

Comment:
Our migrations currently on use `Declaration`, but there's at least one upcoming migration which needs `AtRule`, and the others were trivial to support, so I just threw them all in 👍

Reply:
Great idea 👍 

I've made that change.